CMakeLists.txt

```diff
@@ -540,6 +551,7 @@ endif()
 
 if(HWLOC_FOUND)
   set(DEFAULT_ENABLE_HWLOC ON CACHE INTERNAL "default hwloc")
+  include_directories(${HWLOC_INCLUDE_DIRS})
```

Comment:
This adds the include dir for the entire project, and also it's unclear to me why this is necessary. The cmake code in `lib/CL/devices/topology/CMakeLists.txt` should automatically take care of the pkg-config include directory by the `target_link_libraries( ... PRIVATE PkgConfig::HWLOC)`. Any fixes if necessary should be done there.

Reply:
Ah, thanks, it looks like the `if(PkgConfig::HWLOC)` in `lib/CL/devices/topology/CMakeLists.txt` is not working. I'll look into fixing that and fold the change in https://github.com/pocl/pocl/pull/1664